modules/mfa/src/Auth/Process/Mfa.php

```diff
@@ -719,6 +719,7 @@ protected function redirectToMfaPrompt(array &$state, string $employeeId, array
 
         $state['mfaOptions'] = $mfaOptions;
         $state['managerEmail'] = self::getMaskedManagerEmail($state);
+        $state['unmaskedManagerEmail'] = self::getManagerEmail($state);
```

Comment:
Do you know if the state can be decoded on the client side?

Reply:
I believe the state is exclusively server-side. Data from it is used to render the HTML returned to the client, but I think it's only stored in the session on the server, not sent to the client.

Inspecting the response from the server lines up with that understanding: none of the state data seems to have arrived at the client other than what was intentionally included in the HTML.

Thanks for checking.